modules/base/manifests/ssh_user.pp

```diff
@@ -1,56 +1,42 @@
 define base::ssh_user (
-    $ssh_user,
-    $ssh_password,
-    $ssh_group,
-)
-{
-  #$key          = 'AAAAB3NzaC1yc2EAAAADAQABAAABAQC+R3QXANAoW/mu6yQufC2ZO/llJp7nV1Dk4hi0v2KkyRAk3fQKPgk8FuvuNLxA0IbEMebnQPeEdu3IybJZ9flc8C7DPvB6pP8JHGpS9dZnUjjtVR7ebBJ2dEZcSbJYeP4xrXmu8nCb5EFrvceLRZcBtSwdhtNRBflya6NEFEGee5cVCSmDYfDlapPQOCqIdvTWqHxghQojA5qyQ4psfRfX0+H/wEHGs1ztw4rp+vVlAVLLPsNeOFyq+x6YKNcHloZDpawiT6Qm4DgC3KeUKwfUFLl4GaDZhcu/HFmzmGQqG/YzoUct7zyudcr9w9YHpkV8QgM53wP1alFw3G4EJa7L'
-  #$group        = 'wheel',
-  $shell        = "/bin/bash"
-  $home         = "/home/${ssh_user}"
-  $ssh          = "$home/.ssh"
-  $key          = "$ssh/authorized_keys"
-
+    $ssh_user     = 'admin',
+    $ssh_password = 'MHFqAb1XfnC8Q',
+    $key          = 'AAAAB3NzaC1yc2EAAAADAQABAAABAQCanKzx6z+UImiT3sWUhTrqtWS7WtoyDzv4d48u+f+D0nZlJwP1nXUL3ca32Kz9JbKMLe+MlvZQzW7wJ8m2BVRuKXAIxmLRW3rhYjHgULrSr4OZ46i4Gi/YhDTnyW/28SuOpGD3wlk5cN0rxT0paB/Icoa6nIOjzn+RNJFAvMpblV4O/cVstLJZ2xrZlX+ZoCl3J2OXBkiDc+0xxJU7SEaIkz+WqJceELCKQ76r5GNZXyP+ow6t8egrTJXnBOguo1I8Ssf4UlZ/OnYGsg/17VtmrYulimvZiyJV4ZTmcRbnu1EggQz/2buif8ec2zekhlnjgqN4Ltkwp+D3SYehL0SX'
```

Comment:
I would better pass the path to the .pub key & than extract key from the file

Reply:
puppet.com/docs
ssh_authorized_key { 'nick@magpie.example.com':
  ensure => present,
  user   => 'nick',
  type   => 'ssh-rsa',
  key    => 'AAAAB3Nza[...]qXfdaQ==',
}
Key begins with 'AAA' and ends with '==' but not in your code - is it really works?

Reply:
Yes, Passing such long string as params isn't safe. Imagine that I passed  string with single quote " ' ".